src/Annotation/AccessType.php

```diff
@@ -10,11 +10,27 @@
  *
  * @author Johannes M. Schmitt <schmittjoh@gmail.com>
  */
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::TARGET_PROPERTY)]
 final class AccessType
 {
     /**
      * @Required
-     * @var string
+     * @var string|null
      */
     public $type;
+
+    public function __construct(array $values = [], ?string $type = null)
+    {
+        if ([] !== $values) {
+            if (array_key_exists('value', $values)) {
+                $type = $values['value'];
+            }
+
+            if (array_key_exists('type', $values)) {
+                $type = $values['type'];
+            }
+        }
+
+        $this->type = $type;
```

Comment:
2 suggestions here:


### By keeping the local variable:

```suggestion
        if (array_key_exists('value', $values)) {
            $type = $values['value'];
        }
    
        if (array_key_exists('type', $values)) {
            $type = $values['type'];
        }

        $this->type = $type;
```

### Removing extra local variable:
```suggestion
        if (array_key_exists('value', $values)) {
            $this->type = $values['value'];
        }
    
        if (array_key_exists('type', $values)) {
            $this->type = $values['type'];
        }
```

I propose the 2 versions as I don't know your coding rules about it.

I could try to find some time to make a proposal that removes all `if ([] !== $values)` as they are not required, following your coding rules, if you want. Just tell me which proposal is OK to you.

Thanks

Reply:
I'd go with the first suggestion, as the second one wouldn't work if attributes are disabled

Reply:
Thank you. Do you want me to remove the unecessary `if`s or do you want to do it? I don't want to block you in your work here, creating conflicts for no reason.

Reply:
I've applied the suggestion ;)